src/query/interpret/eval.hpp

```diff
@@ -270,30 +270,28 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
   }
 
   TypedValue Visit(InListOperator &in_list) override {
-    TypedValue *_list_ptr = nullptr;
     TypedValue _list;
     auto literal = in_list.expression1_->Accept(*this);
 
-    auto get_list_literal = [this, &in_list, &_list, &_list_ptr]() -> void {
+    auto get_list_literal = [this, &in_list]() -> TypedValue {
```

Comment:
Doing a copy could be fine. Just think about the types of objects that we could be copying. 
If they are large it could be a good idea to do a reference. That way we avoid a copy in case `ReferenceExpressionEvaluator` is valid.